File: runtime-go/event_helpers.go
package runtime

import (
	"context"
)

// Message represents a generic payload sent or received from an event queue (Kafka, RabbitMQ, etc.).
type Message struct {
	Key       []byte
	Value     []byte
	Topic     string
	Timestamp int64
}

// EventProducer defines a clean interface for dispatching streaming events.
type EventProducer interface {
	Produce(ctx context.Context, msg Message) error
	Close() error
}

// EventConsumer defines a clean interface for subscribing to event streaming topics.
type EventConsumer interface {
	Consume(ctx context.Context, topic string, handler func(msg Message) error) error
	Close() error
}

File: runtimes/python/helix_rt/event_helpers.py
class Message:
    def __init__(self, key: bytes, value: bytes, topic: str, timestamp: int = 0):
        self.key = key
        self.value = value
        self.topic = topic
        self.timestamp = timestamp

class EventProducer:
    """Interface wrapper for dispatching streaming events (Kafka/RabbitMQ/etc.)."""
    async def produce(self, message: Message) -> None:
        raise NotImplementedError

    async def close(self) -> None:
        raise NotImplementedError

class EventConsumer:
    """Interface wrapper for subscribing to event streaming topics (Kafka/RabbitMQ/etc.)."""
    async def consume(self, topic: str, handler) -> None:
        raise NotImplementedError

    async def close(self) -> None:
        raise NotImplementedError
